internal/core/docker_host.go

```diff
@@ -56,6 +56,23 @@ func DefaultGatewayIP() (string, error) {
 	return ip, nil
 }
 
+// defaultCallbackCheckFn Use a vanilla Docker client to check if the Docker host is reachable.
+// It will avoid recursive calls to this function.
+var defaultCallbackCheckFn = func(ctx context.Context, host string) error {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithHost(host), client.WithAPIVersionNegotiation())
+	if err != nil {
+		return err
```

Comment:
suggestion: wrap the error, so we can understand the source.
```suggestion
		return fmt.Errorf("new client: %w", err)
```

Reply:
Done